apps/server/src/schema/integrations/insights-operations.ts

```diff
@@ -50,69 +46,14 @@ builder.queryFields((t) => ({
     type: GroupedInsightsDto,
     nullable: false,
     args: {
-      filter: t.arg({ type: FilterInsightsInput, required: true }),
+      filter: t.arg({ type: FilterInsightsInputDto, required: true }),
     },
     resolve: async (_root, args, ctx, _info) => {
-      const redisValue = await getInsightsCache<GroupedInsightsType>(ctx.organizationId, args.filter);
-      if (redisValue) return redisValue;
-
-      const groupBy: (InsightsColumnsGroupByType | 'currency')[] = [...(args.filter.groupBy ?? []), 'currency'];
-
-      const insightsRaw: Record<string, never>[] = await prisma.$queryRawUnsafe(
-        groupedInsights(args.filter, ctx.organizationId, ctx.acceptedLocale),
+      return getInsightsHelper(
+        args.filter,
+        ctx.organizationId,
+        Language[ctx.acceptedLocale.toUpperCase() as keyof typeof Language],
```

Comment:
```suggestion
        ctx.acceptedLocale
```
We don't like the `as` because it is not gurantees types during run time. Here either we force the acceptedLocale in an enum from the very start. So [here](https://github.com/adsviewer/turboviewer/blob/81cdf8045ab50f80fc1f0d15d5d37867f30bc061/apps/server/src/language.ts#L18), we have a list of all possible languages or we keep it a string. For now let's keep it a string.

In general is better to type things as early as possible in the flow.